### ISSUE TITLE
Rule S121: False Positives in case of Pattern Matching (C# 7.0) 

### ISSUE THREAD

**Issue by ArchutM** [SonarSource/sonar-csharp#1444]
### Description
The rule S121 (Control structures should use curly braces ) reports false positives.
Those false positives follow the Pattern Matching syntax that was introduced in C# 7.0

### Repro steps
Use the pattern matching syntax like:

```
if (o is int i) 
{
    doSomething(i);
}
```
### Expected behavior
There is no finding.

### Actual behavior
There is a false positive finding telling:
_Add curly braces around the nested statement(s) in this "if" block._
This is reported although there are curly braces around 
``` doSomething(i); ```. Obviously the ```i``` leads to the finding.

* SonarC# Version 5.4
* Visual Studio Version: 15.7.3

**Comment by Evangelink** [comment]
Hi @ArchutM,

I cannot reproduce with the latest version of SonarC# so I will close this ticket. Please feel free to re-open or comment if you are still experiencing any issue.
